src/components/common/Extensions/ExtensionDetails/ExtensionDetails.tsx

```diff
@@ -65,19 +71,19 @@ const ExtensionDetails = () => {
     breadCrumbs.push(MSG.setup);
   }
 
+  const hasRegisteredProfile = !!user;
   // @TODO: Extend these checks to include permissions, account and network interaction
   const canExtensionBeUninstalled = !!(
+    hasRegisteredProfile &&
     isInstalledExtensionData(extensionData) &&
     extensionData.uninstallable &&
     extensionData.isDeprecated
   );
   const canExtensionBeDeprecated =
+    hasRegisteredProfile &&
     isInstalledExtensionData(extensionData) &&
     extensionData.uninstallable &&
     !extensionData.isDeprecated;
-  // @TODO: Check if the latest network extension version is greater than the current version
-  const canExtensionBeUpgraded =
-    !!user?.profile && isInstalledExtensionData(extensionData);
```

Comment:
Moved this check to `ExtensionUpgradeButton` as it didn't seem necessary to do it here